tests/utils/state_locks_filter.py

```diff
@@ -0,0 +1,391 @@
+    def _current_lock(self, start_line: int) -> list[str]:
+        next_match = get_next_match(self.lines, start_line, self.LOCK_PREFIX)
+        if next_match == -1:
+            return self.lines[start_line:len(self.lines)]
+        return self.lines[start_line:next_match]
+
+    def get_name(self) -> str:
+        return self.header
+
+    def get_locks(self) -> list[LockOp]:
+        return self.locks
+
+    def get_traces(self) -> list[LockOpTrace]:
+        traces = []
+        for lock in self.locks:
+            traces.append(lock.get_trace())
+        return traces
+
+    def get_lock_held_time(self, trace: LockOpTrace) -> int:
+        for lock in self.locks:
+            if lock.get_trace() == trace:
+                return lock.get_held_time()
+
+        return 0
+
+    def get_lock_wait_time(self, trace: LockOpTrace) -> int:
+        for lock in self.locks:
+            if lock.get_trace() == trace:
+                return lock.get_wait_time()
+
+        return 0
+
+
+class GroupTimes:
+    """
+    GroupTimes: A tuple-like class that associates the group name with the held and wait times for each lock trace.
+    """
+
+    def __init__(self, group_name: str, held_time: int, wait_time: int):
+        self.group_name = group_name
+        self.held_time = held_time
+        self.wait_time = wait_time
+
+    def get_group_name(self) -> str:
+        return self.group_name
+
+    def get_held_time(self) -> int:
+        return self.held_time
+
+    def get_wait_time(self) -> int:
+        return self.wait_time
+
+
+class LockTraceManager:
+    """
+    LockTraceManager: Handles filtering and managing the lock traces and associated group times.
+    It provides methods to filter the traces by time and to print the results in a sorted manner.
+    """
+
+    def __init__(self, traces: dict[LockOpTrace, list[GroupTimes]]):
+        self.traces = traces
+
+    # Filter the times for each trace
+    def filter(self, held_time: int, wait_time: int):
+        filtered_traces = dict[LockOpTrace, list[GroupTimes]]()
+        for trace, times in self.traces.items():
+            filtered_times = [
+                time
+                for time in self.traces.get(trace)
+                if time.get_held_time() >= held_time
+                and time.get_wait_time() >= wait_time
+            ]
+            if len(filtered_times) > 0:
+                filtered_traces[trace] = filtered_times
+
+        self.traces = filtered_traces
+
+    # Keep the traces that match with the params
+    def match(self, match_names: list[str]):
+        filtered_traces = dict[LockOpTrace, list[GroupTimes]]()
+        for trace, times in self.traces.items():
+            for match_name in match_names:
+                if trace.match(match_name):
+                    filtered_traces[trace] = times
+
+        self.traces = filtered_traces
+
+    # print the traces with their times for each test
+    def print(self, sort_held_time, sort_wait_time):
```

Comment:
I think this would pass as type annotation.

```suggestion
    def print(self, sort_held_time: bool, sort_wait_time: bool) -> None:
```